src/DependencyInjection/Configuration.php

```diff
@@ -17,12 +17,20 @@ class Configuration implements ConfigurationInterface
 {
     public function getConfigTreeBuilder()
     {
-        $treeBuilder = new TreeBuilder();
-        $rootNode = $treeBuilder->root('knpu_oauth2_client');
+        $treeBuilder = new TreeBuilder('knpu_oauth2_client');
+        $rootNode = method_exists($treeBuilder, 'getRootNode') ? $treeBuilder->getRootNode() : $treeBuilder->root('knpu_oauth2_client');
 
         $rootNode
             ->children()
             ->scalarNode('http_client')->defaultNull()->info('Service id of HTTP client to use (must implement GuzzleHttp\ClientInterface)')->end()
+            ->arrayNode('http_client_options')
```

Comment:
We need some validation to prevent you from setting both `http_client` and `http_client_options`.

Reply:
I think we still need this one part. I think you may need to add some validation to the parent node - here's an example I think will work https://github.com/symfony/symfony/blob/master/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php#L630-L635